### PR TITLE
Improve help entry of "Mountains" terrain. Resolves #5906

### LINKED ISSUE #5906: Mountain terrain type description - redefining units unable to move to the hex
Section of "mountains" terrain type entry:
> Most cavalry cannot enter mountainous terrain; however, elvish cavalry is an exception to this, as are the goblin wolf riders.

Firstly, there is no mention about Dune Riders being able to climb mountains and even excelling in fighting there, making it yet another exception. At that point land cavalry moving into mountains becomes the norm rather than exception.
Secondly, Heavy Infantry, despite not being cavalry, is unable to move into mountainous terrain.
Thirdly, I think merfolk deserve a mention here as well, as opposed to more mobile naga.

How about replacing that section with:
> Some units simply cannot enter mountainous terrain - most notably Wesnothian-pattern heavy cavalry and shock infantry, burdened by their bulky plate armor, and merfolk, whose piscine body build renders them unable to climb strongly inclined terrain.

### PR DESCRIPTION
**Pull request by max-torch**
Improve help entry of "Mountains" terrain.

### CODE REVIEW

**Comment by Wedge009** [comment]
I have no immediate concerns about this but I'll let others more invested in the nuances of Wesnoth text make the decision on this. Thanks for doing this work, every little bit counts.

File: data/core/terrain.cfg [hunk 1]
```diff
@@ -3100,7 +3100,7 @@ Most units have about 60% defense in a castle."
     string=Mt
     editor_group=rough
     help_topic_text= _ "<italic>text='Mountains'</italic> are steep enough that units often have to climb over obstacles to move.
-By this nature, they provide a considerable defensive bonus for most troops, but they also severely impede any passage through them. Most cavalry simply cannot enter mountainous terrain; however, elvish cavalry is an exception to this, as are the goblin wolf riders. Both dwarves and trolls are native to mountainous terrain, and have a very easy time getting around.
+By this nature, they provide a considerable defensive bonus for most troops, but they also severely impede any passage through them. Units burdened by heavy armor, such as the Wesnoth army's cavalry and heavy infantry lines cannot even enter them. Aquatic races and creatures, such as the merfolk, with their piscine bodies rendering them unable to climb steep inclines, cannot enter mountains as well; however, nagas are an exception due to their many strong arms, rugged scales, and impressive strength. Both dwarves and trolls are native to mountainous terrain, and have a very easy time traversing them.
```
Comment (nemaara) [inline data/core/terrain.cfg:3103]:
```suggestion
By this nature, they provide a considerable defensive bonus for most troops, but also severely impede any passage through them. Units burdened by heavy armor, such as the Wesnoth cavalry and heavy infantry lines, cannot even enter them. Aquatic races and creatures like the merfolk have piscine bodies which also render them unable to climb steep inclines. Nagas are an exception due to their strong arms and rugged scales. Both dwarves and trolls are native to mountainous terrain, and have an easy time traversing them.
```

It's a bit wordy right now, a little more concise would be nicer, and semicolons aren't to be used unless really necessary.

**Review by nemaara — CHANGES_REQUESTED** [review]
